channels/drdynvc/client/drdynvc_main.c

```diff
@@ -1612,46 +1501,34 @@ BOOL VCAPITYPE VirtualChannelEntry(PCHANNEL_ENTRY_POINTS pEntryPoints)
 
 		context->handle = (void*) drdynvc;
 		context->custom = NULL;
-
 		drdynvc->context = context;
-
 		context->GetVersion = drdynvc_get_version;
 		drdynvc->rdpcontext = pEntryPointsEx->context;
-
 		*(pEntryPointsEx->ppInterface) = (void*) context;
 	}
 
 	drdynvc->log = WLog_Get("com.freerdp.channels.drdynvc.client");
-
 	WLog_Print(drdynvc->log, WLOG_DEBUG, "VirtualChannelEntry");
-
-	CopyMemory(&(drdynvc->channelEntryPoints), pEntryPoints, sizeof(CHANNEL_ENTRY_POINTS_FREERDP));
-
+	CopyMemory(&(drdynvc->channelEntryPoints), pEntryPoints,
+	           sizeof(CHANNEL_ENTRY_POINTS_FREERDP));
 	rc = drdynvc->channelEntryPoints.pVirtualChannelInit(&drdynvc->InitHandle,
-		&drdynvc->channelDef, 1, VIRTUAL_CHANNEL_VERSION_WIN2000, drdynvc_virtual_channel_init_event);
+	        &drdynvc->channelDef, 1, VIRTUAL_CHANNEL_VERSION_WIN2000,
+	        drdynvc_virtual_channel_init_event);
 
 	if (CHANNEL_RC_OK != rc)
 	{
 		WLog_ERR(TAG, "pVirtualChannelInit failed with %s [%08X]",
-				 WTSErrorToString(rc), rc);
-		free(drdynvc);
-		free(*(pEntryPointsEx->ppInterface));
-		*(pEntryPointsEx->ppInterface) = NULL;
```

Comment:
This applies to all channels, please review all VirtualChannelEntryPoints where FALSE is returned.